dynamics/wind_model.py

```diff
@@ -0,0 +1,209 @@
+    projection = ((wind_vector[0] * car_vector[0] + wind_vector[1] * car_vector[1]) * car_vector[0],
+                  (wind_vector[0] * car_vector[0] + wind_vector[1] * car_vector[1]) * car_vector[1])
+    parallel_wind_component = projection
+
+    return parallel_wind_component
+
+
+def visualize_vectors(wind_vector, unit_car_vector):
+    """
+    Plots the wind vector, car vector and projected wind vector for visualization, not critical to model functionality
+    @param wind_vector: tuple containing a vector modelling the wind
+    @param unit_car_vector: tuple containing a vector modelling the car
+
+    """
+
+    # Initialize origin
+    x = [0]
+    y = [0]
+
+    # Create unit wind vector
+    wind_vector_magnitude = math.sqrt(wind_vector[0] * wind_vector[0] + wind_vector[1] * wind_vector[1])
+    unit_wind_vector = (wind_vector[0] / wind_vector_magnitude, wind_vector[1] / wind_vector_magnitude)
+
+    projection = (
+        (unit_wind_vector[0] * unit_car_vector[0] + unit_wind_vector[1] * unit_car_vector[1]) * unit_car_vector[0],
+        (unit_wind_vector[0] * unit_car_vector[0] + unit_wind_vector[1] * unit_car_vector[1]) * unit_car_vector[1]
+                )
+    unit_parallel_wind_component = projection
+
+    print("Unit Wind Vector:", unit_wind_vector)
+    print("Unit Car Vector:", unit_car_vector)
+    print("Unit Proj Vector:", unit_parallel_wind_component)
+    plt.quiver(x, y, unit_wind_vector[0], unit_wind_vector[1], color='b', units='xy', scale=1, label="Wind Direction")
+    plt.quiver(x, y, unit_car_vector[0], unit_car_vector[1], color='r', units='xy', scale=1, label="Car Direction")
+    plt.quiver(x, y, unit_parallel_wind_component[0], unit_parallel_wind_component[1], color='g', units='xy', scale=1,
+               label="Projected Wind")
+    plt.title('Vector Plot')
+
+    # set the x-limits and y-limits
+    plt.xlim(-1.5, 1.5)
+    plt.ylim(-1.5, 1.5)
+
+    # Show plot
+    plt.legend(loc="upper left")
+    plt.grid()
+    plt.show()
+
+
+def wind_model_main(coordinates_list):
+    """
+    Calculates the drag of the car for a series of two points, drag is calculated based on wind data at 2nd point
+    @param coordinates_list: list of length 2 containing tuples containing coordinates Ex: [(lat1, long1), (lat2, long2)]
+    @return: Drag force in N
+
+    """
+    # Generate vector modelling car's direction of travel
+    car_vector = gen_car_vector(coordinates_list)
+
+    # Get wind data from API for a given coordinate point
+    current_point = coordinates_list[1]
+
+    # Initialize dataframe to hold weather data
+    headers = ['Latitude', 'Longitude', 'Temperature (C)', 'Wind Speed (m/s)', 'Wind Direction', 'Weather',
+               'Weather Description', 'Pressure (hPa)', 'Precipitation (mm)']
+    weather_df = pd.DataFrame(columns=headers)
+
+    # Get weather data and store into dataframe
+    weather_df = get_weather((current_point[0]), (current_point[1]), weather_df)
```

Comment:
Just a note to self: get_weather is looping + calling on individual points, which is gross. Look for replacement API that can take multiple points maybe?